pages/home.tsx

```diff
@@ -1,11 +1,74 @@
 import Head from 'next/head'
-
+import theme from '../styles/theme'
+import useSWR from 'swr'
 import { withServerSideAuth } from '@clerk/nextjs/ssr'
 import { SSRUser } from '../helpers/user-details'
 import Page from '../components/Page'
 import { ServerPageProps } from '../types/types'
-import { Box, Text } from 'grommet'
+import {
+  Box,
+  Text,
+  Tabs,
+  Tab,
+  Card,
+  CardHeader,
+  CardBody,
+  CardFooter,
+  Button,
+  Layer
+} from 'grommet'
+import { useEffect, useState } from 'react'
+import { PrescriptionAndLocation, PharmacyLocation } from '../types/types'
+import QRCode from 'react-qr-code'
+
 const Home = ({ user }: ServerPageProps) => {
+  const [prescriptions, setPrescriptions] = useState<
+    Array<PrescriptionAndLocation>
+  >([])
+  const [pharmacyLocation, setPharmacyLocation] = useState<PharmacyLocation>()
+  const [location, setLocation] = useState<Location>()
+  const [loading, setLoading] = useState(true)
+  const [qrCode, setQrCode] = useState('')
+  const [showQrCode, setShowQrCode] = useState(false)
+
+  const fetcher = (userRole: string) => {
+    if (userRole === 'patient') {
+      return fetch('/api/prescriptions/read', {
+        method: 'POST',
+        body: JSON.stringify({
+          patientId: user.Patient?.id
+        })
+      })
+        .then((response) => response.json())
+        .then((data) => setPrescriptions(data.prescription))
+        .catch()
+        .finally(() => setLoading(false))
+    } else if (userRole === 'staff') {
+      return fetch('/api/locations/read', {
+        method: 'POST',
+        body: JSON.stringify({
+          id: user.Staff?.locationId
+        })
+      })
+        .then((response) => response.json())
+        .then((data) => {
+          setPharmacyLocation(data.location)
+        })
+        .catch()
+        .finally(() => setLoading(false))
+    }
```

Comment:
All of this code can be restructured. in the `getServerSideProps` you can add a `query` to get the patient Info and staff info see this example on the teams page https://github.com/pharmaboxfydp/pharmabox/blob/9ad02464ce50ab98909831a6b5d64b2fcfdb400a/pages/team.tsx#L71

Reply:
Then you are going to want to restructure this into a react hook using `swr`
see this as an example:
 https://github.com/pharmaboxfydp/pharmabox/blob/main/hooks/team.ts#L29
and then it can be used like so https://github.com/pharmaboxfydp/pharmabox/blob/9ad02464ce50ab98909831a6b5d64b2fcfdb400a/components/TeamMembersTable.tsx#L26